### PR TITLE
#2104056 Implement feedback on challenge 3

### PR DESCRIPTION
**Pull request by Chris-Maina**
#### What does this PR do?
Implements feedback that included:
 * Using dotenv file to store environment variables.
 * Complete documentation of API.
 * Test the whole application and not tests folder.
 * Use documentation to allow the use of API.
 * Resolve pylint errors
 * Catch exceptions in pagination
 * Add decorator function
 * Add price and quantity attribute to shopping items.
 * Add view and update logged in user profile feature.
#### Description of Task to be completed?
 * Pip installed dotenv and defined environment variables in .env file. 
 * Documented shopping items creation, getting, editing and deleting.
 * Wrote tests for init file, changed cover package to app. 
 * Defined testing procedure in README
 * Defined function names in good names variables in pylintrc file
 * Linked apiary documentation to Heroku to allow users to test API
 * Resolved exceptions in page number and limit received from users e.g. negative page numbers.
 * Add decorator function to authenticate users during requests.
 * Changed Shoppingitem model to include price and quantity
 * Include endpoints to handle viewing and updating user profile.

#### How should this be manually tested?
Git clone this repository, cd into the project folder and RUN the command below:
 * nosetests --with-coverage --cover-package=app && coverage report

### CODE REVIEW

File: run.py [hunk 1]
```diff
@@ -1,6 +1,12 @@
...
 import os
 from app import create_app
+from os.path import join, dirname
+from dotenv import load_dotenv
+
+# load .env in the base root
+dotenv_path = join(dirname(__file__), '.env')
+load_dotenv(dotenv_path)
 
 config_name = "development"
```
Comment (kkipyegon) [inline run.py]: Your environment should be defined in your .env

File: tests/test_authentication.py [hunk 2]
```diff
@@ -38,6 +39,14 @@ def test_success_registration(self):
         self.assertIn(
             "You have been registered successfully. Please login", str(res.data))
 
+    def test_user_registration_twice(self):
```
Comment (kkipyegon) [inline tests/test_authentication.py:26]: It is good practice for a particular test to have one assert

File: tests/test_authentication.py [hunk 3]
```diff
@@ -7,6 +7,7 @@
 class UserTestCases(unittest.TestCase):
     """
     Test successful registration
+    Test user registration twice
```
Comment (kkipyegon) [inline tests/test_authentication.py]: Create a Base test so that you don't have to call setup method in each of your test files or test cases

**Comment by kkipyegon** [comment]
Also, check on your variable naming try using underscore to separate two names e.g "slid" should be "sl_id"

File: run.py (hunk 1 above)
Comment (kkipyegon) [reply run.py]: Also on line 16 in run.py is there any reason for specifying the port number?

File: tests/test_authentication.py (hunk 2 above)
Comment (Chris-Maina) [reply tests/test_authentication.py:26]: What do you mean here? I shouldn't have two methods that deal with registration ?

File: tests/test_authentication.py (hunk 3 above)
Comment (Chris-Maina) [reply tests/test_authentication.py]: Okay, let me work on that.

File: run.py (hunk 1 above)
Comment (Chris-Maina) [reply run.py]: Consider this taken care of.

**Pull request merged**